diff --git a/aCis_gameserver/bin/.gitignore b/aCis_gameserver/bin/.gitignore
new file mode 100644
index 0000000..d64c606
--- /dev/null
+++ b/aCis_gameserver/bin/.gitignore
@@ -0,0 +1 @@
+/net/
diff --git a/aCis_gameserver/bin/net/sf/l2j/Config$ClassMasterSettings.class b/aCis_gameserver/bin/net/sf/l2j/Config$ClassMasterSettings.class
index 311b744..a648966 100644
--- a/aCis_gameserver/bin/net/sf/l2j/Config$ClassMasterSettings.class
+++ b/aCis_gameserver/bin/net/sf/l2j/Config$ClassMasterSettings.class
Binary files differ
diff --git a/aCis_gameserver/bin/net/sf/l2j/Config.class b/aCis_gameserver/bin/net/sf/l2j/Config.class
index ae19fbd..8359aa2 100644
--- a/aCis_gameserver/bin/net/sf/l2j/Config.class
+++ b/aCis_gameserver/bin/net/sf/l2j/Config.class
Binary files differ
diff --git a/aCis_gameserver/bin/net/sf/l2j/gameserver/handler/VoicedCommandHandler$SingletonHolder.class b/aCis_gameserver/bin/net/sf/l2j/gameserver/handler/VoicedCommandHandler$SingletonHolder.class
index 8291389..e24e4d8 100644
--- a/aCis_gameserver/bin/net/sf/l2j/gameserver/handler/VoicedCommandHandler$SingletonHolder.class
+++ b/aCis_gameserver/bin/net/sf/l2j/gameserver/handler/VoicedCommandHandler$SingletonHolder.class
Binary files differ
diff --git a/aCis_gameserver/bin/net/sf/l2j/gameserver/handler/VoicedCommandHandler.class b/aCis_gameserver/bin/net/sf/l2j/gameserver/handler/VoicedCommandHandler.class
index cdc1337..8760ee6 100644
--- a/aCis_gameserver/bin/net/sf/l2j/gameserver/handler/VoicedCommandHandler.class
+++ b/aCis_gameserver/bin/net/sf/l2j/gameserver/handler/VoicedCommandHandler.class
Binary files differ
diff --git a/aCis_gameserver/config/server.properties b/aCis_gameserver/config/server.properties
index 1dee21a..6a82dce 100644
--- a/aCis_gameserver/config/server.properties
+++ b/aCis_gameserver/config/server.properties
@@ -185,10 +185,23 @@
 AllowBoat = True
 
 # Allow characters to receive damage from falling.
 EnableFallingDamage = True
 
+#========================================#
+#            Banking System              #
+#========================================#
+
+# To enable banking system set this value to true, default is false.
+BankingEnabled = True
+
+# This is the amount of Goldbars someone will get when they do the .deposit command, and also the same amount they will lose when they do .withdraw
+BankingGoldbarCount = 1
+
+# This is the amount of Adena someone will get when they do the .withdraw command, and also the same amount they will lose when they do .deposit
+BankingAdenaCount = 1000000000
+
 # ================================================================
 #                      Debug, Dev & Test config
 # ================================================================
 
 # Don't load spawns.
diff --git a/aCis_gameserver/java/net/sf/l2j/Config.java b/aCis_gameserver/java/net/sf/l2j/Config.java
index 9ff37ab..7dd7826 100644
--- a/aCis_gameserver/java/net/sf/l2j/Config.java
+++ b/aCis_gameserver/java/net/sf/l2j/Config.java
@@ -193,10 +193,15 @@
 	public static int RIFT_ENTER_COST_CAPTAIN;
 	public static int RIFT_ENTER_COST_COMMANDER;
 	public static int RIFT_ENTER_COST_HERO;
 	public static double RIFT_BOSS_ROOM_TIME_MUTIPLY;
 	
+	/** Banking system */
+	public static boolean BANKING_SYSTEM_ENABLED;
+	public static int BANKING_SYSTEM_GOLDBARS;
+	public static int BANKING_SYSTEM_ADENA;
+	
 	/** Wedding system */
 	public static boolean ALLOW_WEDDING;
 	public static int WEDDING_PRICE;
 	public static boolean WEDDING_SAMESEX;
 	public static boolean WEDDING_FORMALWEAR;
@@ -1131,10 +1136,11 @@
 		ES_SP_BOOK_NEEDED = players.getProperty("EnchantSkillSpBookNeeded", true);
 		DIVINE_SP_BOOK_NEEDED = players.getProperty("DivineInspirationSpBookNeeded", true);
 		SUBCLASS_WITHOUT_QUESTS = players.getProperty("SubClassWithoutQuests", false);
 		ALT_GAME_SUBCLASS_EVERYWHERE = players.getProperty("AltSubclassEverywhere", false);
 		
+		
 		MAX_BUFFS_AMOUNT = players.getProperty("MaxBuffsAmount", 20);
 		STORE_SKILL_COOLTIME = players.getProperty("StoreSkillCooltime", true);
 	}
 	
 	/**
@@ -1192,10 +1198,14 @@
 		
 		AUTO_LOOT = server.getProperty("AutoLoot", false);
 		AUTO_LOOT_HERBS = server.getProperty("AutoLootHerbs", false);
 		AUTO_LOOT_RAID = server.getProperty("AutoLootRaid", false);
 		
+		BANKING_SYSTEM_ENABLED = Boolean.parseBoolean(server.getProperty("BankingEnabled", "false"));
+		BANKING_SYSTEM_GOLDBARS = Integer.parseInt(server.getProperty("BankingGoldbarCount", "1"));
+		BANKING_SYSTEM_ADENA = Integer.parseInt(server.getProperty("BankingAdenaCount", "500000000"));
+		
 		ALLOW_DISCARDITEM = server.getProperty("AllowDiscardItem", true);
 		MULTIPLE_ITEM_DROP = server.getProperty("MultipleItemDrop", true);
 		HERB_AUTO_DESTROY_TIME = server.getProperty("AutoDestroyHerbTime", 15) * 1000;
 		ITEM_AUTO_DESTROY_TIME = server.getProperty("AutoDestroyItemTime", 600) * 1000;
 		EQUIPABLE_ITEM_AUTO_DESTROY_TIME = server.getProperty("AutoDestroyEquipableItemTime", 0) * 1000;
diff --git a/aCis_gameserver/java/net/sf/l2j/gameserver/handler/VoicedCommandHandler.java b/aCis_gameserver/java/net/sf/l2j/gameserver/handler/VoicedCommandHandler.java
index 2ec0a0d..62f7093 100644
--- a/aCis_gameserver/java/net/sf/l2j/gameserver/handler/VoicedCommandHandler.java
+++ b/aCis_gameserver/java/net/sf/l2j/gameserver/handler/VoicedCommandHandler.java
@@ -2,18 +2,20 @@
 
 import java.util.HashMap;
 import java.util.Map;
 
 import net.sf.l2j.gameserver.handler.voicecommandhandlers.DropInfo;
+import net.sf.l2j.gameserver.handler.voicecommandhandlers.BankingCmd;
 
 public class VoicedCommandHandler
 {
 	private final Map<String, IVoicedCommandHandler> _entries = new HashMap<>();
 	
 	protected VoicedCommandHandler()
 	{
 		registerHandler(new DropInfo());
+		registerHandler(new BankingCmd());
 	}
 	
 	private void registerHandler(IVoicedCommandHandler handler)
 	{
 		for (String id : handler.getVoicedCommandList())
diff --git a/aCis_gameserver/java/net/sf/l2j/gameserver/handler/voicecommandhandlers/BankingCmd.java b/aCis_gameserver/java/net/sf/l2j/gameserver/handler/voicecommandhandlers/BankingCmd.java
new file mode 100644
index 0000000..2e9745f
--- /dev/null
+++ b/aCis_gameserver/java/net/sf/l2j/gameserver/handler/voicecommandhandlers/BankingCmd.java
@@ -0,0 +1,80 @@
+/*
+* This program is free software: you can redistribute it and/or modify it under
+* the terms of the GNU General Public License as published by the Free Software
+* Foundation, either version 3 of the License, or (at your option) any later
+* version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+* details.
+*
+* You should have received a copy of the GNU General Public License along with
+* this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+package net.sf.l2j.gameserver.handler.voicecommandhandlers;
+
+import net.sf.l2j.Config;
+import net.sf.l2j.gameserver.handler.IVoicedCommandHandler;
+import net.sf.l2j.gameserver.model.actor.Player;
+import net.sf.l2j.gameserver.network.serverpackets.ItemList;
+
+/**
+* This class trades Gold Bars for Adena and vice versa.
+* 
+* @author Ahmed
+*/
+public class BankingCmd implements IVoicedCommandHandler
+{
+	private static final String[] VOICED_COMMANDS =
+	{
+		"bank", "withdraw", "deposit"
+	};
+
+
+@Override
+public void useVoicedCommand(String command, Player player, String target)
+{
+
+	if(command.equalsIgnoreCase("bank"))
+	{
+		player.sendMessage(".deposit (" + Config.BANKING_SYSTEM_ADENA + " Adena = " + Config.BANKING_SYSTEM_GOLDBARS + " Goldbar) / .withdraw (" + Config.BANKING_SYSTEM_GOLDBARS + " Goldbar = " + Config.BANKING_SYSTEM_ADENA + " Adena)");
+	}
+	else if(command.equalsIgnoreCase("deposit"))
+	{
+		if(player.getInventory().getInventoryItemCount(57, 0) >= Config.BANKING_SYSTEM_ADENA)
+		{
+			player.getInventory().reduceAdena("Goldbar", Config.BANKING_SYSTEM_ADENA, player, null);
+			player.getInventory().addItem("Goldbar", 3470, Config.BANKING_SYSTEM_GOLDBARS, player, null);
+			player.getInventory().updateDatabase();
+			player.sendPacket(new ItemList(player, true));
+			player.sendMessage("Thank you, now you have " + Config.BANKING_SYSTEM_GOLDBARS + " Goldbar(s), and " + Config.BANKING_SYSTEM_ADENA + " less adena.");
+		}
+		else
+		{
+			player.sendMessage("You do not have enough Adena to convert to Goldbar(s), you need " + Config.BANKING_SYSTEM_ADENA + " Adena.");
+		}
+	}
+	else if(command.equalsIgnoreCase("withdraw"))
+	{
+		if(player.getInventory().getInventoryItemCount(3470, 0) >= Config.BANKING_SYSTEM_GOLDBARS)
+		{
+			player.getInventory().destroyItemByItemId("Adena", 3470, Config.BANKING_SYSTEM_GOLDBARS, player, null);
+			player.getInventory().addAdena("Adena", Config.BANKING_SYSTEM_ADENA, player, null);
+			player.getInventory().updateDatabase();
+			player.sendPacket(new ItemList(player, true));
+			player.sendMessage("Thank you, now you have " + Config.BANKING_SYSTEM_ADENA + " Adena, and " + Config.BANKING_SYSTEM_GOLDBARS + " less Goldbar(s).");
+		}
+		else
+		{
+			player.sendMessage("You do not have any Goldbars to turn into " + Config.BANKING_SYSTEM_ADENA + " Adena.");
+		}
+	}
+	return;
+}
+@Override
+public String[] getVoicedCommandList()
+{
+	return VOICED_COMMANDS;
+}
+}
\ No newline at end of file
